nodejs/eks/cluster.ts

```diff
@@ -79,9 +79,9 @@ export interface CoreData {
     eksNodeAccess?: k8s.core.v1.ConfigMap;
     kubeconfig?: pulumi.Output<any>;
     vpcCni?: VpcCni;
-    tags?: { [key: string]: string };
+    tags?: pulumi.Input<{ [key: string]: string }>;
```

Comment:
I think we want `pulumi.Input<{ [key: string]: pulumi.Input<string> }>` throughout.

Reply:
The gist of this change per https://github.com/pulumi/pulumi-eks/issues/146, is that the **key**, not just the value of the map, needs to be an `Input` type, as we need to use an `Output` / Promise (the cluster name) as part of the key: e.g. ```[`k8s.io/cluster-autoscaler/${clusterName}`]: "true"```.

To address this comment and @lukehoban's [comment above](https://github.com/pulumi/pulumi-eks/pull/157#discussion_r292551872), I was thinking of exporting a new type `export type KeyMap = { [key: string]: string };` and changing all the tags to have a definition of `pulumi.Input<KeyMap>` to ensure proper typing of the string k/v map. Does this make sense to everyone?

Reply:
> The gist of this change per #146, is that the key, not just the value of the map, needs to be an Input type, as we need to use an Output / Promise (the cluster name) as part of the key: e.g. [`k8s.io/cluster-autoscaler/${clusterName}`]: "true".

Yes, I know--hence `pulumi.Input<{ [key: string]: pulumi.Input<string> }>` rather than `pulumi.Input<{ [key: string]: string }>` or `{ [key: string]: pulumi.Input<string> }`.

> To address this comment and @lukehoban's comment above, I was thinking of exporting a new type export type KeyMap = { [key: string]: string }; and changing all the tags to have a definition of pulumi.Input<KeyMap> to ensure proper typing of the string k/v map. Does this make sense to everyone?

Something like

```
export type InputTags = pulumi.Input<{ [key: string]: pulumi.Input<string> }>
```

could be useful, yes. I thought we already had something like this in `@pulumi/aws`...

Reply:
Went with @pgavlin's suggestion here ^ - thanks pat!